source/MaterialXTest/ShaderValid.cpp

```diff
@@ -303,22 +309,43 @@ static mx::GlslValidatorPtr createValidator(bool& orthographicView, const std::s
 //
 // Outputs error log if validation fails
 //
-static void runValidation(const std::string& outputPath, const std::string& shaderName, mx::ElementPtr element,
-                          mx::GlslValidatorPtr validator, mx::ShaderGeneratorPtr shaderGenerator,
-                          bool orthographicView, mx::DocumentPtr doc, std::ostream& log, bool outputMtlxDoc=false)
+static void runValidation(const std::string& shaderName, mx::ElementPtr element, mx::GlslValidatorPtr validator,
+                          mx::ShaderGeneratorPtr shaderGenerator, bool orthographicView, mx::DocumentPtr doc,
+                          std::ostream& log, bool outputMtlxDoc=true, const std::string& outputFolder="",
+                          const std::string& outputPath=".")
 {
     mx::SgOptions options;
 
     if(element && doc)
     {
         log << "------------ Run validation with element: " << element->getName() << std::endl;
 
-        std::string shaderPath =  mx::FilePath(outputPath) / shaderName;
+        std::string shaderPath;
+        if (!outputFolder.empty())
+        {
+            std::string outputFolderPath = mx::FilePath(outputPath) / mx::FilePath(outputFolder);
+            // Note: mkdir will fail if the directory already exists which is ok.
+#if defined(_WIN32)
+            _mkdir(outputFolderPath.c_str());
+#else
+            mkdir(outputFolderPath.c_str(), 0777);
+#endif
+            shaderPath = mx::FilePath(outputFolderPath) / mx::FilePath(shaderName);
+        }
+        else
+        {
+            shaderPath = mx::FilePath(outputPath) / mx::FilePath(shaderName);
+        }
+
         mx::ShaderPtr shader = shaderGenerator->generate(shaderName, element, options);
         mx::HwShaderPtr hwShader = std::dynamic_pointer_cast<mx::HwShader>(shader);
-        REQUIRE(hwShader != nullptr);
-        REQUIRE(hwShader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
-        REQUIRE(hwShader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
+        CHECK(hwShader != nullptr);
+        if (hwShader == nullptr)
+        {
+            return;
+        }
+        CHECK(hwShader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
+        CHECK(hwShader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
 
         if (outputMtlxDoc)
```

Comment:
Maybe this variable should change name as it doesn't output a mtlx document but rather performs a render to texture and save out that image.